rust/agama-server/src/manager/web.rs

```diff
@@ -163,3 +167,36 @@ async fn installer_status(
     };
     Ok(Json(status))
 }
+
+/// Returns agama logs
+#[utoipa::path(get, path = "/api/manager/logs", responses(
+  (status = 200, description = "Download logs blob.")
+))]
+
+pub async fn download_logs() -> impl IntoResponse {
+    let path = generate_logs().await;
+    let Ok(path) = path else {
+        return (StatusCode::INTERNAL_SERVER_ERROR).into_response();
+    };
+
+    match ServeFile::new(path)
+        .try_call(Request::new(axum::body::Body::empty()))
+        .await
+    {
+        Ok(res) => res.into_response(),
+        Err(_) => (StatusCode::INTERNAL_SERVER_ERROR).into_response(),
+    }
+}
+
+async fn generate_logs() -> Result<String, Error> {
+    let random_name: String = Alphanumeric.sample_string(&mut rand::thread_rng(), 8);
+    let path = format!("/run/agama/logs_{random_name}");
+
+    Command::new("agama")
+        .args(["logs", "store", "-d", path.as_str()])
+        .status()
+        .expect("Cannot generate logs");
```

Comment:
I think that if the call fails the client will not get any response. Or am I missing something?